config/config.yaml

```diff
@@ -11,25 +11,25 @@ git_url: https://github.com/openshift-pipelines/pipeline-service.git
 # git_ref refers to the git repo's ref to be considered as the source of truth for Argo CD applications.
 git_ref: main
 
-# crs_to_sync is a list of the CRs which need to be synced.
-crs_to_sync:
-  - deployments.apps
-  - services
-  - ingresses.networking.k8s.io
-  - networkpolicies.networking.k8s.io
-  - pipelines.tekton.dev
-  - pipelineruns.tekton.dev
-  - tasks.tekton.dev
-  - repositories.pipelinesascode.tekton.dev
+kcp:
+  # crs_to_sync is a list of the CRs which need to be synced.
+  crs_to_sync:
+    - deployments.apps
+    - services
+    - ingresses.networking.k8s.io
+    - networkpolicies.networking.k8s.io
+    - pipelines.tekton.dev
+    - pipelineruns.tekton.dev
+    - tasks.tekton.dev
+    - repositories.pipelinesascode.tekton.dev
+  # Override the version from config/dependencies.yaml
```

Comment:
I think it would be less confusing for the user to set values in one file before running ckcp script. By doing this (having to set kcp version from another file but rest of the values from here), it will only cause confusion from the user's perspective.

Reply:
I think @bnallapeta point also is that `config/dependencies.yaml` is only used by devenv, so if you comment out the version there will be no place for ops to fetch kcp version.
By the way, and unrelated to this PR, the location of "config/dependencies.yaml" is confusing, it should rather be unedr dev directory

Reply:
I think there's a misunderstanding.

1. The kcp version for the project is set by `config/dependencies.yaml` (e.g. when building images).
2. By default ckcp will use that version.
3. If the user needs to change ckcp behavior, the only file they need to touch is `config/config.yaml`. Maybe my comment is misleading, but what it is trying to say is "if you want ckcp to use a different version, uncomment the line below and define the version you want to use".
4. Regarding the fact that `config/dependencies.yaml` is used only by `devenv`: the goal is to use it (or an upcoming replacement `common/env/dependencies.env`) to be used by anything that requires a version to have a single source of truth.

Does that clear things up for everyone?

Reply:
1. In all of the images, KCP_VERSION is explicitly set to a version - 0.9.0. See [this](https://github.com/openshift-pipelines/pipeline-service/blob/main/images/kcp-registrar/Dockerfile#L34) for example.
2. By default ckcp uses the version from [config/config.yaml](url) file. Not from ckcp/dependencies.yaml. See [this](https://github.com/openshift-pipelines/pipeline-service/blob/main/ckcp/openshift_dev_setup.sh#L20) and [this](https://github.com/openshift-pipelines/pipeline-service/blob/main/ckcp/openshift_dev_setup.sh#L161).
3. This is what I addressed in my initial comment. That the user should only change the version in config.yaml
4. Single source of truth - depends on what is the requirement for it. As of now, there are two files I see that's related to this.
  - config/config.yaml - a yaml where users can modify values that will be used by ckcp.
  - DEPENDENCIES.md - a readme where users can see the version details of every single tool we use.
Why do we need to create another 'single source of truth' file?

Reply:
@Roming22 Could you PTAL at this today? I believe PR 39 on kcp-stable is blocked due to this. 

P.S.: This will probably need a rebase due to the restructuring of our repo now.

Reply:
@bnallapeta Rebase done, and I've removed the bit that was causing trouble. It will come back at a later date ;)